juniorguru_chick/lib/threads.py

```diff
@@ -23,20 +24,41 @@ async def fetch_starting_message(thread: discord.Thread) -> discord.Message | No
     except discord.errors.NotFound:
         return None
 
+def name_thread(message: discord.Message, name_template) -> str | None:
+
+    brackets_regex = re.compile(r"""
+        ^\[     # starts with [
+        .*      # any number of any characters
+        [^\s]   # not a whitespace
+        [^\]]   # not a closing bracket
+        \]      # ends with ]
+        """, re.VERBOSE)
+
+    """If the message includes text in square brackets, use that as name for the thread. Otherwise, use the name template."""
+    if (match:= re.match(brackets_regex, message.content)) is not None:
```

Comment:
I thought it works like this:
```suggestion
    if (match:= brackets_regex.match(message.content)) is not None:
```
But if it works, then I don't mind! I learned something new :)

Reply:
Yeah, that's what I gathered from https://docs.python.org/3/library/re.html#re.match 
But the test pass with your approach, too. Backwards compatibility?